src/plugins/workspace/public/components/workspace_form/utils.ts

```diff
@@ -446,5 +447,55 @@ export const getNumberOfChanges = (
   ) {
     count++;
   }
+  if (
+    newFormData.permissionSettings &&
+    initialFormData.permissionSettings &&
+    convertPermissionsToPrivacyType(newFormData.permissionSettings) !==
+      convertPermissionsToPrivacyType(initialFormData.permissionSettings)
+  ) {
```

Comment:
This number of changes will be failed when `permissionSettings` not exists. How about refactor with code below:
```

    convertPermissionsToPrivacyType(newFormData.permissionSettings ?? []) !==
      convertPermissionsToPrivacyType(initialFormData.permissionSettings ?? [])
```

Reply:
ok